pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/periodictask/ControllerPeriodicTask.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.controller.helix.core.periodictask;
+
+import com.linkedin.pinot.controller.helix.core.PinotHelixResourceManager;
+import com.linkedin.pinot.core.periodictask.BasePeriodicTask;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The base periodic task for pinot controller only. It uses <code>PinotHelixResourceManager</code> to determine
+ * which table resources should be managed by this Pinot controller.
+ */
+public abstract class ControllerPeriodicTask extends BasePeriodicTask {
+  public static final Logger LOGGER = LoggerFactory.getLogger(ControllerPeriodicTask.class);
+  protected final PinotHelixResourceManager _pinotHelixResourceManager;
+  private boolean _amILeader;
+  private static final int DEFAULT_INITIAL_DELAY_IN_SECOND = 120;
+
+  public ControllerPeriodicTask(String taskName, long runFrequencyInSeconds,
+      PinotHelixResourceManager pinotHelixResourceManager) {
+    this(taskName, runFrequencyInSeconds, DEFAULT_INITIAL_DELAY_IN_SECOND, pinotHelixResourceManager);
+  }
+
+  public ControllerPeriodicTask(String taskName, long runFrequencyInSeconds, long initialDelayInSeconds,
+      PinotHelixResourceManager pinotHelixResourceManager) {
+    super(taskName, runFrequencyInSeconds, initialDelayInSeconds);
+    _pinotHelixResourceManager = pinotHelixResourceManager;
+    _amILeader = false;
+  }
+
+  @Override
+  public void init() {
+  }
+
+  @Override
+  public void run() {
+    if (!_pinotHelixResourceManager.isLeader()) {
+      skipLeaderTask();
+    } else {
+      List<String> allTableNames = _pinotHelixResourceManager.getAllTables();
+      processLeaderTask(allTableNames);
+    }
+  }
+
+  private void skipLeaderTask() {
+    if (_amILeader) {
+      onBecomeNotLeader();
+    }
+    _amILeader = false;
+    printSkipMessage();
```

Comment:
As long as we have the task name, we can make printSkipMessage() a private method, with the task name being printed.